x-pack/plugins/uptime/e2e/journeys/utils.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Page } from '@elastic/synthetics';
+import { byTestId } from './uptime.journey';
+
+export async function waitForLoadingToFinish({ page }: { page: Page }) {
+  let isLoadingVisible = true;
+
+  while (isLoadingVisible) {
+    const loading = await page.$(byTestId('kbnLoadingMessage'));
+    isLoadingVisible = loading !== null;
+    await page.waitForTimeout(5 * 1000);
```

Comment:
Is this function going to eventually time out on its own? i.e. is there some kind of external retry count mechanism? It looks like it's only being used in synthetics journeys, so I'm assuming Synthetics takes care of that for us.